h2o-webserver-iface/src/main/java/water/webserver/iface/WebServer.java

```diff
@@ -0,0 +1,12 @@
+package water.webserver.iface;
+
+/**
+ * All the functionality that we need to call on an existing instance of HTTP server (servlet container).
+ */
+public interface WebServer {
```

Comment:
The possibility for any exception to be thrown somewhere on the lower levels places the burden of exception hadling to the caller. I propose to define an exception or set of exceptions with more narrow scope. IOException comes to my mind at first.

This forces at keast some degree error handling inside the implementations. IOException has a `Throwable cause` field, allowing us to contain the original cause inside it.

Reply:
yes, `IOException` might be good, I will wrap these methods - I just didn't want to introduce a new one